Packs/ApiModules/Scripts/MicrosoftGraphMailApiModule/MicrosoftGraphMailApiModule.py

```diff
@@ -0,0 +1,1809 @@
+                file_attachments_result.append(
+                    {
+                        'size': file_size,
+                        'data': file_data,
+                        'name': file_name,
+                        'isInline': is_inline,
+                        'requires_upload': True,
+                        'contentId': attach_id
+                    }
+                )
+        return file_attachments_result
+
+    @staticmethod
+    def upload_attachment(
+        upload_url: str, start_chunk_idx: int, end_chunk_idx: int, chunk_data: bytes, attachment_size: int
+    ) -> requests.Response:
+        """
+        Upload an attachment to the upload URL.
+        Args:
+            upload_url (str): upload URL provided when running 'get_upload_session'
+            start_chunk_idx (int): the start of the chunk file data.
+            end_chunk_idx (int): the end of the chunk file data.
+            chunk_data (bytes): the chunk data in bytes from start_chunk_idx to end_chunk_idx
+            attachment_size (int): the entire attachment size in bytes.
+        Returns:
+            Response: response indicating whether the operation succeeded. 200 if a chunk was added successfully,
+                201 (created) if the file was uploaded completely. 400 in case of errors.
+        """
+        chunk_size = len(chunk_data)
+        headers = {
+            "Content-Length": f'{chunk_size}',
+            "Content-Range": f"bytes {start_chunk_idx}-{end_chunk_idx - 1}/{attachment_size}",
+            "Content-Type": "application/octet-stream"
+        }
+        demisto.debug(f'uploading session headers: {headers}')
+        return requests.put(url=upload_url, data=chunk_data, headers=headers)
+
+    def _get_root_folder_children(self, user_id, overwrite_rate_limit_retry=False):
+        """
+        Get the root folder (Top Of Information Store) children collection.
+
+        :type user_id: ``str``
+        :param user_id: Mailbox address
+
+        :raises: ``Exception``: No folders found under Top Of Information Store folder
+
+        :return: List of root folder children
+        rtype: ``list``
+        """
+        root_folder_id = 'msgfolderroot'
+        if children := self._get_folder_children(user_id, root_folder_id, overwrite_rate_limit_retry):
+            return children
+
+        raise DemistoException("No folders found under Top Of Information Store folder")
+
+    def _get_folder_children(self, user_id, folder_id, overwrite_rate_limit_retry=False):
+        """
+        Get the folder collection under the specified folder.
+
+        :type user_id ``str``
+        :param user_id: Mailbox address
+
+        :type folder_id: ``str``
+        :param folder_id: Folder id
+
+        :return: List of folders that contain basic folder information
+        :rtype: ``list``
+        """
+        return self.http_request('GET',
+                                 f'users/{user_id}/mailFolders/{folder_id}/childFolders?$top=250',
```

Comment:
The number 250 should come for a global (allowing it to be changed from outside) - this will also make it more transparent that we're getting only the top 250 child folders.

Reply:
done